File: 2-pattrons-de-structures/bridge/code-typescript/abstraction/concreteAbstraction/advancedRemoteControl.ts
import { Device } from "../../implementations/interface.js";
import { RemoteControle } from "../remoteControle.js";

export class AdvancedRemoteControl extends RemoteControle {
  constructor(device: Device) {
    super(device);
  }

  mute(): void {
    this.device.setVolume(0);
  }
}

File: 2-pattrons-de-structures/bridge/code-java/abstractions/concrete_abstractions/AdvancedRemoteControl.java
package abstractions.concrete_abstractions;

import abstractions.RemoteControle;
import implementations.*;

public class AdvancedRemoteControl extends RemoteControle {
    public AdvancedRemoteControl(Device device) {
        super(device);
    }

    public void mute() {
        device.setVolume(0);
    }
}